src/Dictionaries/HashedDictionary.h

```diff
@@ -261,6 +262,16 @@ class HashedDictionary final : public IDictionary
 
     PaddedPODArray<Key> getIds() const;
 
+    /// Preallocates the hashtable based on query progress
+    /// (Only while loading all data).
+    ///
+    /// @see preallocate
+    template <typename T>
+    void resize(Attribute & attribute, size_t added_rows);
+    void resize(size_t added_rows);
+    std::atomic<uint64_t> new_size = 0;
```

Comment:
I fell like we don't need a new member-field here.
We only need to calculate size before `blockToAttributes`. Just add optional `size_hint` parameter to `blockToAttributes`.

Reply:
>I fell like we don't need a new member-field here.

Agree

>We only need to calculate size before blockToAttributes. Just add optional size_hint parameter to blockToAttributes.

Don't think that extra parameter is useful for it, since there is std::atomic with exchange, will just call `resize()` before `blockToAttributes()`